File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        if (nums.length === 2) return [0, 1];
	
	    const seen = {};
	    for (let i = 0; i < nums.length; i++) {
		const remainder = target - nums[i];
		if (seen[remainder] !== undefined) {
			return [seen[remainder], i];
		}
		seen[nums[i]] = i;
	}

    }
}

File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l = 0
        r = 0
        n = len(nums)-1
        while r <= n:
            while l <= n and nums[l] != val:
                l+=1
            r = l
            while r <= n and nums[r] == val:
                r+=1
            if r <= n:
                nums[l] = nums[r]
                nums[r] = val
                r = l
        i = 0
        while i <= n and nums[i] != val:
            i+=1
        return i
